src/main/java/application/rest/v1/CustomResourceWatcher.java

```diff
@@ -137,9 +148,56 @@ private static void setLoggerLevel(JsonObject o) {
                             if (level.equals("all"))
                                 Logger.setLogLevel(Logger.LogLevel.ALL);    
                         }
+                        
+                        //check for ui setting
+                        JsonElement levelUi  = loggingObj.get("ui"); 
+                        if (UI_LOG_LEVEL_API != null && levelUi != null && levelUi.isJsonPrimitive())  {               
```

Comment:
minor nit. would have expected the whole block to be surrounded in `if (UI_LOG_LEVEL_API != null) {}` since we won't be doing anything with the UI setting if there's no UI.

Reply:
fixed.